File: PingSysInventory/PingSysInventory.Web/Modules/ServerTypes/Inventory/CategoriesForm.ts
import { BooleanEditor, initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface CategoriesForm {
    CategoryName: StringEditor;
    IsActive: BooleanEditor;
}

export class CategoriesForm extends PrefixedContext {
    static readonly formKey = 'Inventory.Categories';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriesForm.init) {
            CategoriesForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(CategoriesForm, [
                'CategoryName', w0,
                'IsActive', w1
            ]);
        }
    }
}

File: PingSysInventory/PingSysInventory.Web/Modules/Inventory/Forms/CategoriesForm.cs
using PingSysInventory.Inventory;

namespace PingSysInventory.Inventory.Forms;

[FormScript("Inventory.Categories")]
[BasedOnRow(typeof(CategoriesRow), CheckNames = true)]
public class CategoriesForm
{
    public string CategoryName { get; set; }
    public bool IsActive { get; set; }
}
